plugins/nf-float/src/main/com/memverge/nextflow/FloatTaskHandler.groovy

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2013-2023, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.memverge.nextflow
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import nextflow.executor.GridTaskHandler
+import nextflow.processor.TaskRun
+/**
+ * Float task handler
+ */
+@Slf4j
+@CompileStatic
+class FloatTaskHandler extends GridTaskHandler {
+
+    FloatTaskHandler( TaskRun task, FloatGridExecutor executor ) {
+        super(task, executor)
+    }
+
+    /**
+     * Override the grid task handler to never create the task directory.
+     */
+    @Override
+    protected ProcessBuilder createProcessBuilder() {
+
+        // -- log the submit command
+        final cli = executor.getSubmitCommandLine(task, wrapperFile)
+        log.trace "start process ${task.name} > cli: ${cli}"
+
+        // -- prepare submit process
+        return new ProcessBuilder()
+            .command( cli as String[] )
+            .redirectErrorStream(true)
```

Comment:
See my changes in [this commit](https://github.com/MemVerge/nf-float/pull/43/commits/7adf0359d93ac13f9dffb59c93b8b1cb716f23e7#) for this file. If the user is using NFS instead of S3, then the task directory still needs to be created locally (I think).

Reply:
Because with NFS the work directory needs to be mounted where Nextflow is launched right?

Reply:
Interesting, I am not sure why and when I rollback your changes.  I am adding them back.